src/WorkspaceRegistration/WorkspaceRegistration.tsx

```diff
@@ -0,0 +1,299 @@
+            const email = formValues['E-mail Address'];
+            const subject = 'Registration Access Request for Workspace';
+
+            let contents = `Request ID: ${data.request_id}\nEnvironment: ${hostname}`;
+
+            Object.entries(formValues).filter(([key]) => !key.includes('_doNotInclude')).forEach((entry) => {
+              const [key, value] = entry;
+              contents = contents.concat(`\n${key}: ${value}`);
+            });
+            createKayakoTicket(subject, fullName, email, contents, kayakoConfig?.kayakoDepartmentId).then(() => setFormSubmissionStatus({ status: 'success' }),
+              (err) => setFormSubmissionStatus({ status: 'error', text: err.message }));
+          } else {
+            // eslint-disable-next-line no-console
+            console.error('Requestor returns 201 but no request_id in payload body'); // shouldn't get here
+          }
+        } else if (status === 409) {
+          // there is already a request for this user on this study, display a message and disable the button
+          setFormSubmissionButtonDisabled(true);
+          setFormSubmissionStatus({ status: 'warning', text: 'There is already a pending request for workspace access, please wait while we are processing your request.' });
+        } else {
+          // something has gone wrong
+          setFormSubmissionStatus({ status: 'error', text: `Failed to create a request with error code: ${status}. Please try again later. If the error persists, please contact us for help.` });
+        }
+        setReqAccessRequestPending(false);
+      },
+    );
+  };
+
+  const onFinish = (values) => {
+    setReqAccessRequestPending(true);
+    handleRegisterFormSubmission(values);
+  };
+
+  const onReset = () => {
+    form.resetFields();
+  };
+
+  const onRadioChange = (e) => {
+    setRole(e.target.value);
+  };
+
+  if (formSubmissionStatus) {
+    return (
+      <div className='workspace-reg-container'>
+        <div className='workspace-reg-form-container'>
+          {(formSubmissionStatus.status === 'success') ? (
+            <Result
+              status={formSubmissionStatus.status}
+              title='Your access request has been submitted!'
+              subTitle='Thank you for your submission. Requests take up to 1 business day to complete. You will be notified of the status.'
+              extra={[
+                <Link key='discovery' to={'/discovery'}>
+                  <Button>Go To Discovery Page</Button>
+                </Link>,
+              ]}
+            />
+          ) : (
+            <Result
+              status={formSubmissionStatus.status}
+              title='A problem has occurred during submitting the request!'
+              subTitle={formSubmissionStatus.text}
+              extra={[
+                <Button type='primary' key='close' onClick={() => { setFormSubmissionStatus(null); setReqAccessRequestPending(false); }}>
+                  Close
```

Comment:
Does it needs a close button?

Reply:
I didn't see any harm in keeping the button there. It displays the error message and takes back the user to the form again, which seemed intuitive thing to do. Do you have any better suggestions?

Reply:
Apologies. I misread. Probably shouldn't review codes when 🍼